### DISCUSSION TITLE
Migrate Custom Labels into their own Meta Files

### DISCUSSION THREAD

**Discussion by mcarvin8** [forcedotcom/cli#2635] [Ideas]
Currently all custom labels are tracked in this config file - `force-app/main/default/labels/CustomLabels.labels-meta.xml`

Retrieving `CustomLabels` with a wildcard will add all labels in the org to this file and retrieving `CustomLabel` with specific labels will overwrite this file with just those labels specified. 

It would be nice for each custom label to be tracked into their own meta file to help maintain version control over labels. My team continuously has to resolve merge conflicts on the custom labels file in our Git repo and we have seen cases of users overwriting other custom labels based on how they retrieved and the state of their sandboxes.

I developed a custom repo here which shows off this concept (https://github.com/mcarvin8/labels-and-workflows) but it would be nice if this was the default structure for custom labels (and workflows but that can be a different ask) to help the community if they also run into issues similar to ours.

Ideal Label Name Format - `force-app/main/default/labels/{fullName}.labels-meta.xml`

**Comment by daveespo** [comment]
Take a look at #2544  -- that topic addresses the pros/cons of a potential structural change to the project format to support requests like these

**Reply by mcarvin8** [reply]
Thanks for the heads-up! I'll track that topic.
